Trello-y68yUirY updated management core service

marcosraimondi1 opened the pull request:
Updated management core service to match current system status. Added security integration.

marcosraimondi1 committed:
Trello-y68yUirY updated management core service

rozagerardo commented on src/main/resources/application.yml:
```diff
@@ -6,10 +6,10 @@ server:
     mongodb:
-      uri: ${MONGO_DB_URI:mongodb://localhost:27017/a6-projectsmanagement}
-      auto-index-creation: true
+      uri: ${A6_PROJECTS_CORE_MONGO_DB_URI:mongodb://localhost:27017/}${A6_PROJECTS_CORE_MONGO_DB_NAME:a6-projects-management}${A6_PROJECTS_CORE_MONGO_DB_PARAMS:?}
```
Let's update the env variables, replacing `PROJECTS_CORE` for `PROJECTS_MGMT_CORE` of `MGMT_CORE`

rozagerardo commented on src/main/resources/security.yml:
```diff
@@ -0,0 +1,6 @@
+    oauth2:
+      resource-server:
+        jwt:
+          issuer-uri: ${A6_PROJECTS_CORE_OAUTH_SECURITY_ISSUER_URI:http://localhost:9081}
```
Same here. Let's check that  all the env variables contain `MGMT_CORE` and not for other services

rozagerardo commented on src/main/resources/application.yml:
```diff
@@ -6,10 +6,10 @@ server:
-    import: classpath:/api.yml
+    import: classpath:/api.yml, classpath:/security.yml
   application:
     name: ${APP_NAME:projects.management}
```
Since we're already making changes here, let's update the application name to `projects.management.core`...Not critical, I don't think we're using this for anything, but just for correctness

rozagerardo commented on src/test/kotlin/com/angorasix/projects/management/core/presentation/handler/ProjectsPresentationHandlerUnitTest.kt:
```diff
@@ -252,7 +252,7 @@ class ProjectsManagementHandlerUnitTest {
     fun `Given existing projects - When get project for Admin Contributor - Then handler retrieves Ok Response with Edit link`() =
```
Let's update the class/file name from `ProjectsPresentationHandlerUnitTest` to <main handler class name>Unit test: `ProjectsManagementHandlerUnitTest`

rozagerardo commented on src/main/kotlin/com/angorasix/projects/management/core/presentation/handler/ProjectsManagementHandler.kt:
```diff
@@ -267,7 +267,8 @@ private fun ProjectManagementDto.resolveHypermedia(
     add(getByProjectIdAffordanceLink)
 
     // edit ProjectManagement
-    if (simpleContributor != null && simpleContributor.isProjectAdmin) {
```
created Trello task to implement this logic correctly:
https://trello.com/c/bmSXmaFq/18-replicate-admins-channges-in-mgmtcore

(so, no need to change anything here, we'll work on that as a separate task)

marcosraimondi1 committed:
Update application.yml

Updated application name to .core
Updated env variables to reference management environment

marcosraimondi1 committed:
Update security.yml

updated env variables to contain MGMT_CORE and not for other services

marcosraimondi1 committed:
Rename ProjectsPresentationHandlerUnitTest.kt to ProjectsManagementHandlerUnitTest.kt
